SAP/spartacus
# feat: CXSPA-5903 Schematics - add `provideHttpClient(withFetch(), withInterceptorsFromDi())` in fresh applications

This PR contains changes related usage of `provideHttpClient(withFetch(), withInterceptorsFromDi())` in new Spartacus applications as well as in our internal storefront application. Such a change allows us to avoid warning during Spartacus installation:
```
NG02801: Angular detected that `HttpClient` is not configured to use `fetch` APIs. It's strongly recommended to enable
`fetch` for applications that use Server-Side Rendering for better performance and compatibility. To enable `fetch`, add the
'withFetch()` to the `provideHttpClient()` call at the root of the application.
```
This change is applied for both client-side rendering and server-side rendering.

In addition to `withFetch` config, `withInterceptorsFromDi` has to be applied to include class-based interceptors configured using a multi-provider in the injector into the configured HttpClient instance.

Changes:
- `provideHttpClient(withFetch())` instead of `HttpClientModule` in fresh Spartacus applications
- adjust the schematics logic responsible for removing `provideClientHydration` to remove only the intended provider
- `provideHttpClient(withFetch())` instead of `HttpClientModule` in the storefront app module
- unit test adjustments

QA steps:

- CSR:
  - build and deploy Spartacus packages locally:
  `npx ts-node ./tools/schematics/testing.ts` in the SPA root folder
  - generate fresh ng17 application:
  `npx @angular/cli new my-app --standalone=false --ssr=false --style=scss`
  - go to the project's directory install Spartacus from local packages:
  `npx @angular/cli add @spartacus/schematics@latest --baseUrl="https://40.76.109.9:9002"`
  - verify whether `app.module.ts` contains `provideHttpClient(withFetch(), withInterceptorsFromDi())` configuration
  - run `npm run start` to verify if the app works
  
- SSR:
  - build and deploy Spartacus packages locally:
  `npx ts-node ./tools/schematics/testing.ts` in the SPA root folder
  - generate fresh ng17 application:
  `npx @angular/cli new my-app --standalone=false --ssr=false --style=scss`
  - go to the project's directory and install Spartacus with SSR from local packages:
  `npx @angular/cli add @spartacus/schematics@latest --baseUrl="https://40.76.109.9:9002" --ssr`
  - verify whether `app.module.ts` contains `provideHttpClient(withFetch(), withInterceptorsFromDi())` configuration (the additional coma is related to the removal of `provideClientHydration` feature)
  - run `npm run build` and `npm run serve:ssr:my-app` to verify if the app works
  
closes [CXSPA-5903](https://jira.tools.sap/browse/CXSPA-5903)
